cpp/tests/io/json/json_test.cpp

```diff
@@ -2776,4 +2776,29 @@ TEST_F(JsonReaderTest, JSONMixedTypeChildren)
   }
 }
 
+TEST_F(JsonReaderTest, MixedTypesWithSchema)
+{
+  std::string data = "{\"data\": {\"A\": 0, \"B\": 1}}\n{\"data\": [1,0]}\n";
+
+  std::map<std::string, cudf::io::schema_element> data_types;
+  data_types.insert(
+    std::pair{"data", cudf::io::schema_element{cudf::data_type{cudf::type_id::LIST}}});
+
```

Comment:
I am a little confused by this test. The data type of the schema element is just LIST? but the test I posed in the bug https://github.com/rapidsai/cudf/issues/15260#issue-2176843293 has it as a LIST of STRING. I'm not really sure if that makes much of a difference, but it makes a big difference if I want a LIST<STRUCT<a:STRING,b:STRING>> compared to just a LIST<STRING>